### DISCUSSION TITLE
Welcome to tor-browser-selenium Discussions!

### DISCUSSION THREAD

**Discussion by gunesacar** [webfp/tor-browser-selenium#139] [General]
## 👋 Welcome!
  Feel free to ask questions and share ideas here.

**Comment by gunesacar** [comment]
You can find common errors and potential solutions in the Troubleshooting section:  
https://github.com/webfp/tor-browser-selenium#-troubleshooting

**Comment by SaWaib** [comment]
Im trying to use tbselenium to click on elements on a website, i managed to get element.click() working but only if i scroll to the element myself. if not i get the error:"element could not be scrolled into view". is there a command to automatically scroll to an element or a different way to solve this?

**Reply by gunesacar** [reply]
This sounds like a selenium-related question. Perhaps you can find help in Selenium documentation or forums.

**Comment by TiagoGuerra001** [comment]
Hey i wanted to use this but not using tor Network so i can do some studies about the the security option and how it restrains UX and i need to not use Tor network any ideia what i need to adapt?

**Reply by gunesacar** [reply]
You can disconnect the computer from the Internet and launch Tor browser with tbselenium. Would that help?

**Reply by TiagoGuerra001** [reply]
I wanted to visit websites just no using the tor network and check what broke under different security sliders and wanted to automate that process using this

**Reply by gunesacar** [reply]
Got it. Then the best approach would be to accomplish what you described manually with a Tor Browser (without any automation). Then apply the same settings/prefs using tbselenium. I can help with the latter.
Also see: https://superuser.com/questions/1117383/can-i-use-tor-browser-without-using-tor-network (likely, outdated)

**Reply by TiagoGuerra001** [reply]
ok so i done i have Tor browser without tor and everything is working fine the part thats left using the tbselenium tool

**Reply by TiagoGuerra001** [reply]
Hey, so im not understanding how to do it. Could you give me some insights so i can try to make this work?

**Reply by gunesacar** [reply]
Can you describe what you changed  in Tor Browser to make it  work without the Tor network? Did you change some settings or preferences? If you share those, I can make suggestions on how to reproduce it with tbselenium.

**Reply by TiagoGuerra001** [reply]
Open about:config and enter each of the followng setting each one to FALSE and the last one to zero.
extensions.torlauncher.start_tor = FALSE
network.dns.disabled = FALSE
network.proxy.socks_remote_dns = FALSE
network.proxy.type = 0
and run in the terminal
env TOR_SKIP_LAUNCH=1 TOR_TRANSPROXY=1 ./start-tor-browser.desktop
i checked in the tor website that says if im using tor or not and it seems to work good

**Reply by TiagoGuerra001** [reply]
so i added the preferences and and the enviroment in the place where you add other preferences and enviroment variables i think that is fine but it still complains about the sock ports not listening
this is the Error
SOCKS port 9050 is not listening

**Reply by TiagoGuerra001** [reply]
so i believe my changes have done nothing since i have enable tor in 9050 and it says its connected to tor networks which is not supposed to do since i add the things i needed

**Reply by gunesacar** [reply]
> so i added the preferences and and the enviroment in the place where you add other preferences and enviroment variables i think that is fine but it still complains about the sock ports not listening

Did you do all these using tbselenium? Can you share a minimal reproducible example?

**Reply by TiagoGuerra001** [reply]
so i have added them in the tbdriver.py file 
set_pref('network.dns.disabled',False)
and so forth and the enviroment variables with this
environ["TOR_SKIP_LAUNCH"] = 1
